base/compiler/ssair/inlining.jl

```diff
@@ -1768,6 +1768,13 @@ function late_inline_special_case!(ir::IRCode, idx::Int, stmt::Expr, flag::UInt3
         unionall_call = Expr(:foreigncall, QuoteNode(:jl_type_unionall), Any, svec(Any, Any),
             0, QuoteNode(:ccall), stmt.args[2], stmt.args[3])
         return SomeCase(unionall_call)
+    elseif f === _typeof_captured_variable
+        if isa(type, Const)
+            return SomeCase(quoted(type.val))
+        elseif isconstType(type)
+            return SomeCase(quoted(type.parameters[1]))
+        end
+        # TODO we may still want to inline the body of `_typeof_captured_variable` here
```

Comment:
Why is this special case necessary? The cases being optimized here should be handled by the handling for `MethodPureResult`.